runtime/lua/vim/fs.lua

```diff
@@ -732,4 +732,28 @@ function M.abspath(path)
   return M.joinpath(cwd, path)
 end
 
+--- Return relative path from directory `base` to `target`. Note that `base` must be an ancestor to
+--- `target`, meaning relative paths starting with `../` are not supported.
+--- Example:
+---
+--- ```lua
+--- vim.fs.relpath('/var', '/var/lib') -- lib
+--- ```
+---
+--- @param base string
+--- @param target string
+--- @return string|nil
+function M.relpath(base, target)
+  vim.validate('base', base, 'string')
+  vim.validate('target', target, 'string')
+
+  base = vim.fs.normalize(vim.fs.abspath(base))
+  target = vim.fs.normalize(vim.fs.abspath(target))
+  if base == target then
+    return ''
```

Comment:
Should this be `'.'`  ?

Reply:
I guess we have to since `vim.fs.joinpath` doesn't support empty strings to denote current directory.